src/Subscriber.js

```diff
@@ -18,10 +18,28 @@ class Subscriber {
 
     this._retryMap = new Map()
 
-    this._callback = () => Promise.resolve()
     this._initializePromise = undefined
+
+    this.actions = new Map()
+  }
+
+  _callback (msg) {
+    let {action, data} = typeof msg === 'object' ? msg : {}
+    if (!this.actions.has(action)) {
+      return Promise.resolve()
+    }
+
+    let handler = this.actions.get(action)
+    return handler(data)
```

Comment:
As I said earlier we could wrap this inside a Promise just to be sure the `_callback` will return a Promise whatever is inside the `handler`. What do you think @tunderdomb ?

The usage currently does not require to return with a Promise but that was the default behavior in earlier versions. So maybe we could normalize and always return with a Promise...

Reply:
fix